packages/nimble-components/src/table-column/text/template-helpers.ts

```diff
@@ -0,0 +1,12 @@
+export const setTitleWhenOverflow = (
+    span: HTMLElement,
+    title: string
+): void => {
+    if (title && span.offsetWidth < span.scrollWidth) {
+        span.setAttribute('title', title);
```

Comment:
We shouldn't be mutating the elements directly and instead rely on template binding.

So imagine writing this from scratch like "normal", it probably would look like:

```ts
/* @internal */
@observable titleOverflow = false;

updateTitleOverflow () {
        this.titleOverflow = this.spanRef.offsetWidth < this.spanRef.scrollWidth;
}

clearTitleOverflow () {
    this.titleOverflow = false;
}
```

And then in the template:

```ts
`
@mouseover="${x => x.updateTitleOverflow()}"
@mouseout="${x => x.clearTitleOverflow()}"
title="${x => x.titleOverflow && x.content ? x.content : undefined}"
`
```

Biggest benefit is that the template stays responsible for binding values to the element, it's much clearer how title is being defined in-case it needs to be updated to reference other state, etc..

Reply:
Should have looked at @msmithNI's representation that molly referenced lol. This example above aligns with that one conceptually, I think this one is a tad clearer with the helper functions.